_posts/2024-10-24-autorebalancing-on-scaling.md

```diff
@@ -0,0 +1,152 @@
+  skipHardGoalCheck: true
+  # ... other rebalancing related configuration
+```
+
+The configuration template can be the same for both scaling operations or the user can decide to use different ones.
+Using the `KafkaRebalance` custom resource is a way to make the lives of Strimzi users easier because they already know how it works in regards to integration with Cruise Control.
+There isn't anything new to learn beyond applying the template annotation and not specifying fields like `mode` and `brokers` because they will be set automatically by the Strimzi operator when running the auto-rebalancing.
+Indeed, when an auto-rebalancing has to run, the Cluster Opertaor starts from the template and creates an actual `KafkaRebalance` custom resource by using that configuration and adding the right `mode` and the `brokers`.
+
+In the following shell snippet, you can see the `KafkaRebalance` resources involved in two auto-rebalancing operations.
+The first one is about scaling up and the next one is about scaling down.
+It shows how the `KafkaRebalance` resources go through the usual states when the operator interacts with Cruise Control for running the rebalancing.
+
+```shell
+NAME                                            CLUSTER      TEMPLATE   STATUS
+my-add-brokers-rebalancing-template             my-cluster   true       
+my-remove-brokers-rebalancing-template          my-cluster   true       
+...
+my-cluster-auto-rebalancing-add-brokers         my-cluster              PendingProposal
+my-cluster-auto-rebalancing-add-brokers         my-cluster              ProposalReady
+my-cluster-auto-rebalancing-add-brokers         my-cluster              Rebalancing
+my-cluster-auto-rebalancing-add-brokers         my-cluster              Ready
+...
+...
+my-cluster-auto-rebalancing-remove-brokers      my-cluster              ProposalReady
+my-cluster-auto-rebalancing-remove-brokers      my-cluster              Rebalancing
+my-cluster-auto-rebalancing-remove-brokers      my-cluster              Ready
+```
+
+It is also possible to get the auto-rebalancing progress from the `status` section within the `Kafka` custom resource as well as from the specific `KafkaRebalance` instance created by the operator of course.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+  cruiseControl:
+    # ...
+    autoRebalance:
+      # ...
+status:
+  autoRebalance:
+    lastTransitionTime: "2024-10-24T10:46:10.759494479Z"
+    modes:
+    - brokers:
+      - 6
+      - 7
+      mode: remove-brokers
+    state: RebalanceOnScaleUp
+```
+
+If you want to see the auto-rebalancing in action, you can watch this [video](https://www.youtube.com/watch?v=u6qHPL-VXRE) published on the official Strimzi YouTube channel.
+
+### Conclusion
+
+Scaling an Apache Kafka cluster plays an important role in getting better performance on one side and saving costs and energy consumption on the other side.
```

Comment:
We could also add that "A better integration between Kafka an Cruise Control is one of the long-term goals of the Strimzi project that would allow a sort of Kafka auto pilot mode, and this is a significant step in that direction.". Or something like this.